docs/tasks/administer-cluster/dns-custom-nameservers.md

```diff
@@ -0,0 +1,150 @@
+---
+assignees:
+- bowei
+- zihongz
+title: Configuring private DNS zones and upstream nameservers in Kubernetes
+---
+
+{% capture overview %}
+This page shows how to add custom private DNS zones (stub domains) and upstream
+nameservers.
+{% endcapture %}
+
+{% capture prerequisites %}
+* {% include task-tutorial-prereqs.md %}
+* The cluster must be configured to use the `kube-dns` addon.
+{% endcapture %}
+
+{% capture steps %}
+
+## Name resolution in Kubernetes
+
+The diagram below shows the flow of DNS queries specified in the configuration
+above. With the dnsPolicy set to “ClusterFirst” a DNS query is first sent to
+the DNS caching layer in kube-dns. From here, the suffix of the request is
+examined and then forwarded to the appropriate DNS.  In this case, names with
+the cluster suffix (e.g.; “.cluster.local”) are sent to kube-dns. Names with
+the stub domain suffix (e.g.; “.acme.local”) will be sent to the configured
+custom resolver. Finally, requests that do not match any of those suffixes will
+be forwarded to the upstream DNS.
+
+![DNS lookup flow](/docs/tasks/administer-cluster/dns-custom-nameservers/dns.png)
+
+## Configuring stub-domain and upstream DNS servers
```

Comment:
Do we need to mention which kube-dns version starts supporting this? May be confusing if user try it on old kube-dns.